src/main/java/com/couchbase/lite/replicator/ChangeTracker.java

```diff
@@ -326,7 +326,8 @@ public void process(HttpRequest request, HttpContext context) throws HttpExcepti
                 Log.v(Log.TAG_CHANGE_TRACKER, "%s: Making request to %s", this, maskedRemoteWithoutCredentials);
                 HttpResponse response = httpClient.execute(request);
                 StatusLine status = response.getStatusLine();
-                if (status.getStatusCode() >= 300 && !Utils.isTransientError(status)) {
+                if ((status.getStatusCode() >= 300 && !Utils.isTransientError(status)) ||
+                        status.getStatusCode() >= 300 && mode != ChangeTrackerMode.LongPoll) {
```

Comment:
2 things:
1. I checked both iOS and .NET and this change will make the code divert from both platform.
2. I understand that this will make continuous pull replicator stops right away without having change tracker retrying.